pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -135,6 +135,37 @@ func createService(componentMeta metav1.ObjectMeta, componentExt *v1beta1.Compon
 			ClusterIP: corev1.ClusterIPNone,
 		},
 	}
+	if val, ok := componentMeta.Labels[constants.ODHKserveRawAuth]; ok && val == "true" {
+		if service.ObjectMeta.Annotations == nil {
+			service.ObjectMeta.Annotations = make(map[string]string)
+		}
+		isvcname := componentMeta.Name
+		if val, ok := componentMeta.Labels[constants.InferenceServiceLabel]; ok {
+			isvcname = val
+		}
+		service.ObjectMeta.Annotations["service.beta.openshift.io/serving-cert-secret-name"] = isvcname
+		httpsPort := corev1.ServicePort{
+			Name: "https",
+			Port: constants.OauthProxyPort,
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.String,
+				StrVal: "https",
+			},
+			Protocol: corev1.ProtocolTCP,
+		}
+		ports := service.Spec.Ports
+		replaced := false
+		for i, port := range ports {
+			if port.Port == constants.CommonDefaultHttpPort {
+				ports[i] = httpsPort
+				replaced = true
+			}
+		}
+		if !replaced {
+			ports = append(ports, httpsPort)
+		}
```

Comment:
From https://github.com/opendatahub-io/kserve/blob/4add27b4fcdc34a2ae4ceb8e50d6838db6ec8651/pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go#L70-L110, looks like the `CommonDefaultHttpPort` entry should be present always, and it is going to be the first entry always. So, I think you can simply mutate `service.Spec.Ports[0]` and change the target port...

...sorry, I know that I said about being defensive in my previous comment.